### PR TITLE
Azure Migrate new cmdlets to support VMwareCbt Migration

### PR DESCRIPTION
**Pull request by prsadhu-ms-idc**
## Description
Design PR : https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/737
Azure Migrate PowerShell cmdlets for the VMware to Azure scenario will enable IT Pro administrators to automate their migration factories and make it easy to perform repetitive migration activities (as most of the migration activities are done in stages). It will also enable at scale automation support for customers (currently limited due to portal constraints), a repetitive ask from our customers. The focus will be on allowing end to end scenario enablement for VMware to Azure scenario by giving more power to the user without exposing the additional complexity inherent in Azure Migrate’s migration architecture.
Enable IT Pros to achieve to automate VMware to Azure agentless migration scenario
Will allow scripting of Azure Migrate migration operations for scale deployments
Equip power users to unleash Azure Migrate’s full potential through simplified customization
Ensure future extensibility in accordance with Azure Migrate migration roadmap

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by prsadhu-ms-idc** [comment]
As discussed by @rahug1190 , raising PR for these custom cmdlets for review. 
New-AzMigrateServerReplication
Set-AzMigrateServerReplication
Start- AzMigrateTestMigration
Start-AzMigrateTestMigrationCleanup
Remove-AzMigrateServerReplication
Restart-AzMigrateServerReplication
Start-AzMigrateServerMigration
Get-AzMigrateServerReplication
Get-AzMigrateJob
New-AzMigrateDiskMapping

File: src/Migrate/docs/Get-AzmigrateJob.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
+---
```
Comment (isra-fel) [inline src/Migrate/docs/Get-AzmigrateJob.md:1]: Wrong casing in file name `Get-Az"m"igrateJob`

File: src/Migrate/examples/Get-AzmigrateJob.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+### Example 1: {{ Add title here }}
```
Comment (isra-fel) [inline src/Migrate/examples/Get-AzmigrateJob.md:1]: Wrong casing in file name `Get-Az"m"igrateJob`

File: src/Migrate/test/Get-AzmigrateJob.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,26 @@
+$loadEnvPath = Join-Path $PSScriptRoot 'loadEnv.ps1'
```
Comment (isra-fel) [inline src/Migrate/test/Get-AzmigrateJob.Tests.ps1:1]: Wrong casing in file name `Get-Az"m"igrateJob`

**Review by isra-fel — COMMENTED** [review]
A few issues before this can be merged:
- cmdlet implemetation is missing `src/Migrate/custom/Initialize-AzMigrateReplicationInfrastructure.ps1`
- Missing examples and test cases
- File name casing
- A lot of cmdlets are removed in the PR such as `*-AzMigrateGroup` and `*-AzMigrateHyperVSite`. Is it intentional?

File: src/Migrate/docs/Get-AzmigrateJob.md (hunk 1 above)
Comment (prsadhu-ms-idc) [reply src/Migrate/docs/Get-AzmigrateJob.md]: Fixed

**Comment by prsadhu-ms-idc** [comment]
cmdlet implemetation is missing src/Migrate/custom/Initialize-AzMigrateReplicationInfrastructure.ps1 - Removed from PR
Missing examples and test cases - Filling them in
File name casing - Removed wrong file name
A lot of cmdlets are removed in the PR such as *-AzMigrateGroup and *-AzMigrateHyperVSite. Is it intentional? - Yes, our swagger contains lots of cmdlets, which is not to be exposed.

**Comment by isra-fel** [comment]
@prsadhu-ms-idc got it. Please click the "Ready for review" button when it's ready.

**Review by VeryEarly — CHANGES_REQUESTED** [review]
please mask subscription in examples and regenerate help

**Review by isra-fel — CHANGES_REQUESTED** [review]
Hi @prsadhu-ms-idc it generally LGTM. One thing is that I noticed you used `-ResourceName` for the name of the resource, however Azure PowerShell has a convention to use just `-Name` for the name of the subject of the cmdlet. Could you please replace all `-ResourceName` with `-Name`?

**Comment by isra-fel** [comment]
I just added Az.Migrate to our CI pipeline. Re-triggering the checkes

**Comment by prsadhu-ms-idc** [comment]
Hi @isra-fel , please find reply.
One thing is that I noticed you used -ResourceName for the name of the resource, however Azure PowerShell has a convention to use just -Name for the name of the subject of the cmdlet. Could you please replace all -ResourceName with -Name?

[Answers] 2 points

-  In Migrate Project name can mean a lot of things, like Fabric Name, Container Name, Policy Name, Site Name, Project Name and then Resource Name. So swagger to remove this ambiguity uses Resource Name. 

- The cmdlets that use them (Get-AzMigrateReplicationPolicy , Get-AzMigrateReplicationProtectionContainer etc.) are not supposed to be exposed to User in long run. Currently we are doing that because there was a cmdlet Initialise-Infrastructure, which would have used them (and we would have kept the cmdlets internal). But because of cross RP build issue, which we are unable to resolve right now, we have kept the cmdlets exposed. These are not to be used by User in flow. We assure you as soon as we fix the cross RP issue in next release of this, and definitely before General Availiablity, we will remove these cmdlets. Right now renaming will be a lot of work, because it is used in many places in custom code. Request you to kindly consider same.

**Comment by prsadhu-ms-idc** [comment]
@VeryEarly for 
please mask subscription in examples and regenerate help
as something like xxx-xxx-xxx

[Answer] Fixed.

**Review by isra-fel — APPROVED** [review]
I got you point. `-Name` should refer to the subject of current cmdlet, but yours are not the case. I'm good to go.

**Pull request merged**